metronome/checkpointing/interpreter/test/src/io/iohk/metronome/checkpointing/interpreter/InterpreterServiceSpec.scala

```diff
@@ -0,0 +1,275 @@
+
+    implicit lazy val interpreterTracer: Tracer[Task, InterpreterEvent] =
+      Tracer.noOpTracer[Task, InterpreterEvent]
+
+    val resources: Resource[Task, Input] =
+      for {
+        (serviceAddress, interpreterAddress) <- Resource.liftF(
+          randomAddressPair
+        )
+        rnd = new SecureRandom()
+        List(serviceKeyPair, interpreterKeyPair) = List.fill(2) {
+          ECKeyPair.generate(rnd)
+        }
+
+        serviceConnection <- makeLocalConnectionManager(
+          localKeyPair = serviceKeyPair,
+          localAddress = serviceAddress,
+          targetKeyPair = interpreterKeyPair,
+          targetAddress = interpreterAddress
+        )
+
+        interpreterConnection <- makeLocalConnectionManager(
+          localKeyPair = interpreterKeyPair,
+          localAddress = interpreterAddress,
+          targetKeyPair = serviceKeyPair,
+          targetAddress = serviceAddress
+        )
+
+        // Wait until the two connection managers establish and settle a single connection.
+        _ <- Resource.liftF {
+          for {
+            // Allow some time for connections to be sorted out.
+            _ <- Task.sleep(retryConfig.oppositeConnectionOverlap)
+            _ <- Task
+              .parSequence(
+                List(
+                  serviceConnection.isConnected.restartUntil(identity),
+                  interpreterConnection.isConnected.restartUntil(identity)
+                )
+              )
+              .timeout(5.seconds)
+          } yield ()
+        }
+
+        // Start processing messages going from the Service to the Interpreter.
+        serviceRpc <- InterpreterService[Task](
+          interpreterConnection,
+          interpreterRpc,
+          timeout = 1.second
+        )
+
+        serviceConsumer <- serviceConnection.incomingMessages.consume
+
+      } yield new Input(serviceRpc, serviceConnection, serviceConsumer)
+  }
+
+  val randomAddressPair: Task[(InetSocketAddress, InetSocketAddress)] = Task {
+    val s1 = new ServerSocket(0)
+    try {
+      val s2 = new ServerSocket(0)
+      try {
+        val a1 = new InetSocketAddress("localhost", s1.getLocalPort)
+        val a2 = new InetSocketAddress("localhost", s2.getLocalPort)
+        a1 -> a2
+      } finally {
+        s2.close()
+      }
+    } finally {
+      s1.close()
+    }
+  }
+
+  def makeLocalConnectionManager(
+      localKeyPair: ECKeyPair,
+      localAddress: InetSocketAddress,
+      targetKeyPair: ECKeyPair,
+      targetAddress: InetSocketAddress
+  ): Resource[Task, LocalConnectionManager[
+    Task,
+    ECPublicKey,
+    InterpreterMessage
+  ]] = {
+    import DefaultInterpreterCodecs.interpreterMessageCodec
+
+    implicit val networkTracers = NetworkTracers(
+      Tracer.noOpTracer[Task, NetworkEvent[ECPublicKey, InterpreterMessage]]
+      // Tracer.instance[Task, NetworkEvent[ECPublicKey, InterpreterMessage]](e =>
```

Comment:
If it's your intention to keep it around, which may be a good idea, then I'd rather precede it with a note, so that it doesn't look accidental. Something like:

```
// Uncomment below to log network events to stdout
```

Reply:
Done.